featurebyte/query_graph/node/nested.py

```diff
@@ -645,4 +639,11 @@ def _derive_on_demand_view_code(
             var_name = var_name_generator.convert_to_variable_name("feat", node_name=self.name)
             expression = get_object_class_from_function_call("pd.to_datetime", expr)
             return [(var_name, expression)], var_name
+        if self.parameters.output_dtype == DBVarType.OBJECT:
+            var_name = var_name_generator.convert_to_variable_name("feat", node_name=self.name)
+            expression = get_object_class_from_function_call(
+                f"{expr}.apply",
+                ExpressionStr("lambda x: np.nan if pd.isna(x) else ast.literal_eval(x)"),
```

Comment:
Is `ast.literal_eval` or `json.loads` more appropriate? I'm under the impression that the dictionaries are encoded as json when returned. Curious the behaviour when the data has special key / values like unicode.

Reply:
This is a good catch!